cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -407,13 +411,76 @@ static void sc_mkdir_and_mount_and_bind(const char *rootfs_dir,
 	}
 }
 
+static int sc_mount_nvidia_is_driver_in_dir(const char *dir)
+{
+	char driver_path[512] = { 0 };
+
+	struct sc_nvidia_driver driver;
+
+	// Probe sysfs to get the version of the driver that is currently inserted.
+	sc_probe_nvidia_driver(&driver);
+
+	// If there's no driver then we should not bother ourselves with finding the
+	// matching library
+	if (driver.major_version == 0) {
+		return 0;
+	}
+	// Probe if a well known library is found in directory dir
+	sc_must_snprintf(driver_path, sizeof driver_path,
+			 "%s/" LIBNVIDIA_GLCORE_SO_PATTERN, dir,
+			 driver.major_version, driver.minor_version);
+
+	if (access(driver_path, F_OK) == 0) {
+		debug("nvidia library detected at path %s", driver_path);
+		return 1;
+	}
+	return 0;
+}
+
 static void sc_mount_nvidia_driver_multiarch(const char *rootfs_dir)
 {
-	// Attempt mount of both the native and 32-bit variants of the driver if they exist
-	sc_mkdir_and_mount_and_bind(rootfs_dir, "/usr/lib/nvidia",
-				    SC_LIBGL_DIR);
-	sc_mkdir_and_mount_and_bind(rootfs_dir, "/usr/lib32/nvidia",
-				    SC_LIBGL32_DIR);
+	const char *native_libdir = NATIVE_LIBDIR "/" HOST_ARCH_TRIPLET;
+	const char *lib32_libdir = NATIVE_LIBDIR "/" HOST_ARCH32_TRIPLET;
+
+	if ((strlen(HOST_ARCH_TRIPLET) > 0) &&
+	    (sc_mount_nvidia_is_driver_in_dir(native_libdir) == 1)) {
+
+		debug("using arch triplet %s", HOST_ARCH_TRIPLET);
+
+		const char *native_sources[] = { native_libdir };
```

Comment:
Since you are creating an array of strings where the array only has 1 element, perhaps a comment is warranted. I suggest here and down below something along the lines of:
```
// sc_mkdir_and_mount_and_glob_files() takes an array of strings, so
// initialize native_sources accordingly, but calculate the array length
// dynamically to make adjustments to native_sources easier.
```